backend/backend_test.go

```diff
@@ -15,20 +15,322 @@
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/addresses"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/blockchain"
+	blockchainMocks "github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/blockchain/mocks"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/btc/types"
 	coinpkg "github.com/digitalbitbox/bitbox-wallet-app/backend/coins/coin"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/eth"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/eth/erc20"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/coins/eth/rpcclient/mocks"
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/config"
+	"github.com/digitalbitbox/bitbox-wallet-app/backend/devices/usb"
 	keystoremock "github.com/digitalbitbox/bitbox-wallet-app/backend/keystore/mocks"
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/keystore/software"
 	"github.com/digitalbitbox/bitbox-wallet-app/backend/signing"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/observable"
+	"github.com/digitalbitbox/bitbox-wallet-app/util/test"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/require"
 )
 
+func mustKeypath(keypath string) signing.AbsoluteKeypath {
+	kp, err := signing.NewAbsoluteKeypath(keypath)
+	if err != nil {
+		panic(err)
+	}
+	return kp
+}
+
+func mustXKey(key string) *hdkeychain.ExtendedKey {
+	xkey, err := hdkeychain.NewKeyFromString(key)
+	if err != nil {
+		panic(err)
+	}
+	return xkey
+}
+
+func keystoreHelper1() *software.Keystore {
+	// From mnemonic: wisdom minute home employ west tail liquid mad deal catalog narrow mistake
+	rootKey := mustXKey("xprv9s21ZrQH143K3gie3VFLgx8JcmqZNsBcBc6vAdJrsf4bPRhx69U8qZe3EYAyvRWyQdEfz7ZpyYtL8jW2d2Lfkfh6g2zivq8JdZPQqxoxLwB")
+	return software.NewKeystore(rootKey)
+}
+
+func keystoreHelper2() *software.Keystore {
+	// From mnemonic: lava scare swap mystery lawsuit army rubber clean mean bronze keen volcano
+	rootKey := mustXKey("xprv9s21ZrQH143K3cfe2832UrUDA5jmFWvm3acoempvZofxin26VdqjosJfTjHsVgjgszDYHiEgepM7J7U9N7HpayNZDRPUoxGKQbJCuHzgnuy")
+	return software.NewKeystore(rootKey)
+}
+
+func rootFingerprint1() []byte {
+	return []byte{0x55, 0x55, 0x55, 0x55}
+}
+
+func rootFingerprint2() []byte {
+	return []byte{0x66, 0x66, 0x66, 0x66}
+}
+
+func rootFingerprint1Func() ([]byte, error) {
+	return rootFingerprint1(), nil
+}
+
+func rootFingerprint2Func() ([]byte, error) {
+	return rootFingerprint2(), nil
+}
+
+// A keystore with a similar config to a BitBox02 Multi - supporting unified and multiple accounts,
+// no legacy P2PKH.
+func makeBitBox02Multi() *keystoremock.KeystoreMock {
+	return &keystoremock.KeystoreMock{
+		NameFunc: func() (string, error) {
+			return "Mock name", nil
+		},
+		RootFingerprintFunc: rootFingerprint1Func,
```

Comment:
below you use ` keystoreHelper1().ExtendedPublicKey`, but why not use ` keystoreHelper1().RootFingerprint` here then too?

Reply:
Nice point! Done. I also overridden it with the fixed `0x55..` value in the tests where we look for specific account codes